app/rolebuilder.go

```diff
@@ -257,10 +259,12 @@ func (rb *roleBuilder) reconcileGlobalRoles(mgmt *config.ManagementContext) erro
 			return false, nil, errors.Errorf("unexpected type comparing %v and %v", have, want)
 		}
 
-		equal := haveGR.DisplayName == wantGR.DisplayName && reflect.DeepEqual(haveGR.Rules, wantGR.Rules)
+		builtin := haveGR.Builtin == wantGR.Builtin
+		equal := haveGR.DisplayName == wantGR.DisplayName && reflect.DeepEqual(haveGR.Rules, wantGR.Rules) && builtin
```

Comment:
nit: builtin can be placed as a first condition. This way if it's false, you save on other conditions' evaluation. DeepEquals in particular